content/tutorial-nlp-from-scratch.md

```diff
@@ -0,0 +1,848 @@
+| Violence against LGBTQA+                         | Michelle Bachelet       | United Nations office of high commisioner official website |
+| I have a dream                                   | Martin Luther King      | Brittanica official website
+
++++
+
+## 2. Preprocess the datasets
+>Preprocessing data is an extremely crucial step before building any Deep learning model, however in an attempt to keep the tutorial focused on building the model, we will not dive deep into the code for preprocessing. Given below is a brief overview of all the steps we undertake to clean our data and convert it to its numeric representation. The [code](https://github.com/Dbhasin1/ethics-tutorial/blob/lstm-update/tutorials/text_preprocessing.py) is public and we encourage you to look it up for a better understanding of this section and can make ethical preprocessing choices in your future work.
+
+1. **Text Denoising** : Before converting your text into vectors, it is important to clean it and remove all unhelpful parts a.k.a the noise from your data by converting all characters to lowercase, removing html tags, brackets and stop words (words that don't add much meaning to a sentence). Without this step the dataset is often a cluster of words that the computer doesn't understand.
+
+
+2. **Converting words to vectors** : A word embedding is a learned representation for text where words that have the same meaning have a similar representation. Individual words are represented as real-valued vectors in a predefined vector space. GloVe is an unsupervised algorithm developed by Stanford for generating word embeddings by generating global word-word co-occurence matrix from a corpus. You can download the zipped files containing the embeddings from https://nlp.stanford.edu/projects/glove/. Here you can choose any of the four options for different sizes or training datasets
+ >The GloVe word embeddings include sets that were trained on billions of tokens, some up to 840 billion tokens. These algorithms exhibit stereotypical biases, such as gender bias which can be traced back to the original training data. For example certain occupations seem to be more biased towards a particular gender, reinforcing problematic stereotypes. The nearest solution to this problem are some de-biasing algorithms as the one presented in https://web.stanford.edu/class/archive/cs/cs224n/cs224n.1184/reports/6835575.pdf which one can use on embeddings of their choice to mitigate bias, if present.
+
++++
+
+You'll start with importing the necessary packages to build our Deep Learning network
+
+```{code-cell} ipython3
+# Importing the necessary packages 
+import numpy as np 
+import pandas as pd 
+import matplotlib.pyplot as plt 
+from text_preprocessing import TextPreprocess 
+import string
+```
+
+````{admonition}**Define a class to perform the aforementioned text preprocessing and conversion techniques on the input data**
```

Comment:
I asked @melissawm about this, it should render when the `.md` file is converted to `.html`

Reply:
I think we talked about the toggle not this one. If we want the notebook to be executable I think we should stick to regular markdown syntax.

Reply:
Just to clarify: if we use the admonition directive, this works fine for the html generated pages, but not for the executable notebook. The [Myst-NB](https://myst-nb.readthedocs.io/en/latest/use/hiding.html) documentation highlights the option of using a cell tag to toggle hiding cells from the html input. This works fine except that in the executable notebook the cell is not hidden. I think this is ok, since if people to download the executable notebook they probably want to see the entire code. So I'd suggest the following: 
- Remove the admonition and `:class: toggle` options, marking this cell as a regular executable python/code cell;
- Edit the cell metadata to add the `hide-input` tag. You can do this in jupyter-lab by clicking the Property Inspector on the right toolbar and adding a tag to the selected cell:
![Screenshot_20210923_163408](https://user-images.githubusercontent.com/3949932/134572309-b40aebab-660a-44b8-8253-117d917369ce.png)

Reply:
Understood, Thanks @melissawm!